### ISSUE TITLE
3.0 alpha13: 02-test_errstr.t is broken on hurd

### ISSUE THREAD

**Issue by sebastianas** [openssl/openssl#14763]
The test suite reports:
```
 02-test_errstr.t ...................
 # The results of this test will end up in test-runs/test_errstr
 1..180
 ../../util/wrap.pl ../../apps/openssl errstr fffffffffffffed2 => 0
 not ok 10 - match 'reason(2147483647)' (18446744073709551314) with one of ( '(ipc/mig) server detected error', 'reason(-302)' )
 
 #   Failed test 'match 'reason(2147483647)' (18446744073709551314) with one of ( '(ipc/mig) server detected error', 'reason(-302)' )'
 #   at ../test/recipes/02-test_errstr.t line 73.
 ../../util/wrap.pl ../../apps/openssl errstr fffffffffffffed4 => 0
 not ok 17 - match 'reason(2147483647)' (18446744073709551316) with one of ( '(ipc/mig) client type check failure', 'reason(-300)' )
 
 #   Failed test 'match 'reason(2147483647)' (18446744073709551316) with one of ( '(ipc/mig) client type check failure', 'reason(-300)' )'
 #   at ../test/recipes/02-test_errstr.t line 73.
 ../../util/wrap.pl ../../apps/openssl errstr fffffffffffffed3 => 0
 not ok 19 - match 'reason(2147483647)' (18446744073709551315) with one of ( '(ipc/mig) wrong reply message ID', 'reason(-301)' )
 
 #   Failed test 'match 'reason(2147483647)' (18446744073709551315) with one of ( '(ipc/mig) wrong reply message ID', 'reason(-301)' )'
 #   at ../test/recipes/02-test_errstr.t line 73.
 ../../util/wrap.pl ../../apps/openssl errstr fffffffffffffed0 => 0
 not ok 31 - match 'reason(2147483647)' (18446744073709551312) with one of ( '(ipc/mig) server type check failure', 'reason(-304)' )
 
 #   Failed test 'match 'reason(2147483647)' (18446744073709551312) with one of ( '(ipc/mig) server type check failure', 'reason(-304)' )'
 #   at ../test/recipes/02-test_errstr.t line 73.
 ../../util/wrap.pl ../../apps/openssl errstr fffffffffffffecb => 0
 not ok 75 - match 'reason(2147483647)' (18446744073709551307) with one of ( '(ipc/mig) destroy request with no reply', 'reason(-309)' )
 
 #   Failed test 'match 'reason(2147483647)' (18446744073709551307) with one of ( '(ipc/mig) destroy request with no reply', 'reason(-309)' )'
 #   at ../test/recipes/02-test_errstr.t line 73.
 ../../util/wrap.pl ../../apps/openssl errstr fffffffffffffed1 => 0
 not ok 119 - match 'reason(2147483647)' (18446744073709551313) with one of ( '(ipc/mig) bad request message ID', 'reason(-303)' )
 
 #   Failed test 'match 'reason(2147483647)' (18446744073709551313) with one of ( '(ipc/mig) bad request message ID', 'reason(-303)' )'
 #   at ../test/recipes/02-test_errstr.t line 73.
 ../../util/wrap.pl ../../apps/openssl errstr fffffffffffffecd => 0
 not ok 132 - match 'reason(2147483647)' (18446744073709551309) with one of ( '(ipc/mig) array not large enough', 'reason(-307)' )
 
 #   Failed test 'match 'reason(2147483647)' (18446744073709551309) with one of ( '(ipc/mig) array not large enough', 'reason(-307)' )'
 #   at ../test/recipes/02-test_errstr.t line 73.
 ../../util/wrap.pl ../../apps/openssl errstr fffffffffffffecc => 0
 not ok 153 - match 'reason(2147483647)' (18446744073709551308) with one of ( '(ipc/mig) server died', 'reason(-308)' )
 
 #   Failed test 'match 'reason(2147483647)' (18446744073709551308) with one of ( '(ipc/mig) server died', 'reason(-308)' )'
 #   at ../test/recipes/02-test_errstr.t line 73.
 ../../util/wrap.pl ../../apps/openssl errstr fffffffffffffece => 0
 not ok 159 - match 'reason(2147483647)' (18446744073709551310) with one of ( '(ipc/mig) server raised exception', 'reason(-306)' )
 
 #   Failed test 'match 'reason(2147483647)' (18446744073709551310) with one of ( '(ipc/mig) server raised exception', 'reason(-306)' )'
 #   at ../test/recipes/02-test_errstr.t line 73.
 ../../util/wrap.pl ../../apps/openssl errstr fffffffffffffecf => 0
 not ok 163 - match 'reason(2147483647)' (18446744073709551311) with one of ( '(ipc/mig) no reply should be sent', 'reason(-305)' )
 
 #   Failed test 'match 'reason(2147483647)' (18446744073709551311) with one of ( '(ipc/mig) no reply should be sent', 'reason(-305)' )'
 #   at ../test/recipes/02-test_errstr.t line 73.
 # Looks like you failed 10 tests of 180.
 Dubious, test returned 10 (wstat 2560, 0xa00)
 Failed 10/180 subtests
…
 Test Summary Report
 -------------------
 02-test_errstr.t                 (Wstat: 2560 Tests: 180 Failed: 10)
   Failed tests:  10, 17, 19, 31, 75, 119, 132, 153, 159
                 163
   Non-zero exit status: 10

```
This looks like a sign-extension error. The good cases are:

```
 ../../util/wrap.pl ../../apps/openssl errstr c0000031 => 0
 ok 76 - match 'Cannot assign requested address' (3221225521) with one of ( 'Cannot assign requested address', 'reason(1073741873)' )
 ../../util/wrap.pl ../../apps/openssl errstr c000001d => 0
 ok 77 - match 'Illegal seek' (3221225501) with one of ( 'Illegal seek', 'reason(1073741853)' )
 ../../util/wrap.pl ../../apps/openssl errstr c000001c => 0
 ok 78 - match 'No space left on device' (3221225500) with one of ( 'No space left on device', 'reason(1073741852)' )
 ../../util/wrap.pl ../../apps/openssl errstr c000002a => 0
 ok 79 - match 'Protocol not available' (3221225514) with one of ( 'Protocol not available', 'reason(1073741866)' )
 ../../util/wrap.pl ../../apps/openssl errstr c0000050 => 0
 ok 80 - match 'Authentication error' (3221225552) with one of ( 'Authentication error', 'reason(1073741904)' )
 ../../util/wrap.pl ../../apps/openssl errstr 90000003 => 0
 ok 81 - match '(ipc/send) invalid destination port' (2415919107) with one of ( '(ipc/send) invalid destination port', 'reason(268435459)' )
 ../../util/wrap.pl ../../apps/openssl errstr c0000023 => 0
 ok 82 - match 'Resource temporarily unavailable' (3221225507) with one of ( 'Resource temporarily unavailable', 'reason(1073741859)' )
 ../../util/wrap.pl ../../apps/openssl errstr c0000071 => 0
 ok 83 - match 'Out of streams resources' (3221225585) with one of ( 'Out of streams resources', 'reason(1073741937)' )

```
A similar issue was reported earlier in https://github.com/openssl/openssl/issues/12276 and closed but I missed that part and not get earlier to retest.

Sebastian

**Comment by levitte** [comment]
What happens if you apply this patch and run that test again?

``` diff
diff --git a/test/recipes/02-test_errstr.t b/test/recipes/02-test_errstr.t
index 6bc07f6d65..70586cd49b 100644
--- a/test/recipes/02-test_errstr.t
+++ b/test/recipes/02-test_errstr.t
@@ -149,5 +149,6 @@ sub match_syserr_reason {
     # The OpenSSL fallback string
     push @strings, "reason($errcode)";
 
+    print STDERR ERR_SYSTEM_FLAG, ' : ', $errcode, "\n";
     return match_opensslerr_reason(ERR_SYSTEM_FLAG | $errcode, @strings);
 }
```

Running an individual test is fairly easy, in this case: `make test TESTS='test_errstr`
If you want more verbosity, add the extra argument `V=1`

**Comment by sebastianas** [comment]
->
 ```
02-test_errstr.t ..
 # The results of this test will end up in test-runs/test_errstr
 1..180
 2147483648 : 22
 ../../util/wrap.pl ../../apps/openssl errstr 80000016 => 0
 ok 1 - match '(os/kern) invalid host' (2147483670) with one of ( '(os/kern) invalid host', 'reason(22)' )
 2147483648 : 1073741877
 ../../util/wrap.pl ../../apps/openssl errstr c0000035 => 0
 ok 2 - match 'Software caused connection abort' (3221225525) with one of ( 'Software caused connection abort', 'reason(1073741877)' )
 2147483648 : 268435461
 ../../util/wrap.pl ../../apps/openssl errstr 90000005 => 0
 ok 3 - match '(ipc/send) will notify' (2415919109) with one of ( '(ipc/send) will notify', 'reason(268435461)' )
 2147483648 : 1073741859
 ../../util/wrap.pl ../../apps/openssl errstr c0000023 => 0
 ok 4 - match 'Resource temporarily unavailable' (3221225507) with one of ( 'Resource temporarily unavailable', 'reason(1073741859)' )
 2147483648 : 3
 ../../util/wrap.pl ../../apps/openssl errstr 80000003 => 0
 ok 5 - match '(os/kern) no space available' (2147483651) with one of ( '(os/kern) no space available', 'reason(3)' )
 2147483648 : 1073741839
 ../../util/wrap.pl ../../apps/openssl errstr c000000f => 0
 ok 6 - match 'Block device required' (3221225487) with one of ( 'Block device required', 'reason(1073741839)' )
 2147483648 : 27
 ../../util/wrap.pl ../../apps/openssl errstr 8000001b => 0
 ok 7 - match 'reason(27)' (2147483675) with one of ( 'Unknown error (os/kern) 27', 'reason(27)' )
 2147483648 : 1073741881
 ../../util/wrap.pl ../../apps/openssl errstr c0000039 => 0
 ok 8 - match 'Transport endpoint is not connected' (3221225529) with one of ( 'Transport endpoint is not connected', 'reason(1073741881)' )
 2147483648 : 1073741833
 ../../util/wrap.pl ../../apps/openssl errstr c0000009 => 0
 ok 9 - match 'Bad file descriptor' (3221225481) with one of ( 'Bad file descriptor', 'reason(1073741833)' )
 2147483648 : 268451848
 ../../util/wrap.pl ../../apps/openssl errstr 90004008 => 0
 ok 10 - match '(ipc/rcv) invalid data' (2415935496) with one of ( '(ipc/rcv) invalid data', 'reason(268451848)' )
 2147483648 : 8
 ../../util/wrap.pl ../../apps/openssl errstr 80000008 => 0
 ok 11 - match '(os/kern) no access' (2147483656) with one of ( '(os/kern) no access', 'reason(8)' )
 2147483648 : 2509
 ../../util/wrap.pl ../../apps/openssl errstr 800009cd => 0
 ok 12 - match '(os/device) read only' (2147486157) with one of ( '(os/device) read only', 'reason(2509)' )
 2147483648 : 1073741891
 ../../util/wrap.pl ../../apps/openssl errstr c0000043 => 0
 ok 13 - match 'Too many processes' (3221225539) with one of ( 'Too many processes', 'reason(1073741891)' )
 2147483648 : 1073741844
 ../../util/wrap.pl ../../apps/openssl errstr c0000014 => 0
 ok 14 - match 'Not a directory' (3221225492) with one of ( 'Not a directory', 'reason(1073741844)' )
 2147483648 : 1073741934
 ../../util/wrap.pl ../../apps/openssl errstr c000006e => 0
 ok 15 - match 'No data available' (3221225582) with one of ( 'No data available', 'reason(1073741934)' )
 2147483648 : 1073741876
 ../../util/wrap.pl ../../apps/openssl errstr c0000034 => 0
 ok 16 - match 'Network dropped connection on reset' (3221225524) with one of ( 'Network dropped connection on reset', 'reason(1073741876)' )
 2147483648 : 268451847
 ../../util/wrap.pl ../../apps/openssl errstr 90004007 => 0
 ok 17 - match '(ipc/rcv) invalid notify port' (2415935495) with one of ( '(ipc/rcv) invalid notify port', 'reason(268451847)' )
 2147483648 : 1073741887
 ../../util/wrap.pl ../../apps/openssl errstr c000003f => 0
 ok 18 - match 'File name too long' (3221225535) with one of ( 'File name too long', 'reason(1073741887)' )
 2147483648 : 1073741896
 ../../util/wrap.pl ../../apps/openssl errstr c0000048 => 0
 ok 19 - match 'RPC struct is bad' (3221225544) with one of ( 'RPC struct is bad', 'reason(1073741896)' )
 2147483648 : 1073741863
 ../../util/wrap.pl ../../apps/openssl errstr c0000027 => 0
 ok 20 - match 'Destination address required' (3221225511) with one of ( 'Destination address required', 'reason(1073741863)' )
 2147483648 : 1073741925
 ../../util/wrap.pl ../../apps/openssl errstr c0000065 => 0
 ok 21 - match 'Translator died' (3221225573) with one of ( 'Translator died', 'reason(1073741925)' )
 2147483648 : 268435463
 ../../util/wrap.pl ../../apps/openssl errstr 90000007 => 0
 ok 22 - match '(ipc/send) interrupted' (2415919111) with one of ( '(ipc/send) interrupted', 'reason(268435463)' )
 2147483648 : 1073741853
 ../../util/wrap.pl ../../apps/openssl errstr c000001d => 0
 ok 23 - match 'Illegal seek' (3221225501) with one of ( 'Illegal seek', 'reason(1073741853)' )
 2147483648 : 268435462
 ../../util/wrap.pl ../../apps/openssl errstr 90000006 => 0
 ok 24 - match '(ipc/send) notify in progress' (2415919110) with one of ( '(ipc/send) notify in progress', 'reason(268435462)' )
 2147483648 : 19
 ../../util/wrap.pl ../../apps/openssl errstr 80000013 => 0
 ok 25 - match '(os/kern) urefs overflow' (2147483667) with one of ( '(os/kern) urefs overflow', 'reason(19)' )
 2147483648 : 1073741856
 ../../util/wrap.pl ../../apps/openssl errstr c0000020 => 0
 ok 26 - match 'Broken pipe' (3221225504) with one of ( 'Broken pipe', 'reason(1073741856)' )
 2147483648 : 4
 ../../util/wrap.pl ../../apps/openssl errstr 80000004 => 0
 ok 27 - match '(os/kern) invalid argument' (2147483652) with one of ( '(os/kern) invalid argument', 'reason(4)' )
 2147483648 : 1073741835
 ../../util/wrap.pl ../../apps/openssl errstr c000000b => 0
 ok 28 - match 'Resource deadlock avoided' (3221225483) with one of ( 'Resource deadlock avoided', 'reason(1073741835)' )
 2147483648 : 2508
 ../../util/wrap.pl ../../apps/openssl errstr 800009cc => 0
 ok 29 - match '(os/device) memory allocation failure' (2147486156) with one of ( '(os/device) memory allocation failure', 'reason(2508)' )
 2147483648 : 1073741937
 ../../util/wrap.pl ../../apps/openssl errstr c0000071 => 0
 ok 30 - match 'Out of streams resources' (3221225585) with one of ( 'Out of streams resources', 'reason(1073741937)' )
 2147483648 : 16
 ../../util/wrap.pl ../../apps/openssl errstr 80000010 => 0
 ok 31 - match '(os/kern) invalid task' (2147483664) with one of ( '(os/kern) invalid task', 'reason(16)' )
 2147483648 : 1073741867
 ../../util/wrap.pl ../../apps/openssl errstr c000002b => 0
 ok 32 - match 'Protocol not supported' (3221225515) with one of ( 'Protocol not supported', 'reason(1073741867)' )
 2147483648 : 1073741902
 ../../util/wrap.pl ../../apps/openssl errstr c000004e => 0
 ok 33 - match 'Function not implemented' (3221225550) with one of ( 'Function not implemented', 'reason(1073741902)' )
 2147483648 : 1073741945
 ../../util/wrap.pl ../../apps/openssl errstr c0000079 => 0
 ok 34 - match 'State not recoverable' (3221225593) with one of ( 'State not recoverable', 'reason(1073741945)' )
 2147483648 : 1073741851
 ../../util/wrap.pl ../../apps/openssl errstr c000001b => 0
 ok 35 - match 'File too large' (3221225499) with one of ( 'File too large', 'reason(1073741851)' )
 2147483648 : 1073741903
 ../../util/wrap.pl ../../apps/openssl errstr c000004f => 0
 ok 36 - match 'Inappropriate file type or format' (3221225551) with one of ( 'Inappropriate file type or format', 'reason(1073741903)' )
 2147483648 : 1073741936
 ../../util/wrap.pl ../../apps/openssl errstr c0000070 => 0
 ok 37 - match 'No message of desired type' (3221225584) with one of ( 'No message of desired type', 'reason(1073741936)' )
 2147483648 : 268435467
 ../../util/wrap.pl ../../apps/openssl errstr 9000000b => 0
 ok 38 - match '(ipc/send) invalid notify port' (2415919115) with one of ( '(ipc/send) invalid notify port', 'reason(268435467)' )
 2147483648 : 1073741944
 ../../util/wrap.pl ../../apps/openssl errstr c0000078 => 0
 ok 39 - match 'Owner died' (3221225592) with one of ( 'Owner died', 'reason(1073741944)' )
 2147483648 : 1073741873
 ../../util/wrap.pl ../../apps/openssl errstr c0000031 => 0
 ok 40 - match 'Cannot assign requested address' (3221225521) with one of ( 'Cannot assign requested address', 'reason(1073741873)' )
 2147483648 : 23
 ../../util/wrap.pl ../../apps/openssl errstr 80000017 => 0
 ok 41 - match '(os/kern) memory present' (2147483671) with one of ( '(os/kern) memory present', 'reason(23)' )
 2147483648 : 1073741866
 ../../util/wrap.pl ../../apps/openssl errstr c000002a => 0
 ok 42 - match 'Protocol not available' (3221225514) with one of ( 'Protocol not available', 'reason(1073741866)' )
 2147483648 : 1073741899
 ../../util/wrap.pl ../../apps/openssl errstr c000004b => 0
 ok 43 - match 'RPC program version wrong' (3221225547) with one of ( 'RPC program version wrong', 'reason(1073741899)' )
 2147483648 : 268435471
 ../../util/wrap.pl ../../apps/openssl errstr 9000000f => 0
 ok 44 - match '(ipc/send) invalid msg-type' (2415919119) with one of ( '(ipc/send) invalid msg-type', 'reason(268435471)' )
 2147483648 : 1073741935
 ../../util/wrap.pl ../../apps/openssl errstr c000006f => 0
 ok 45 - match 'Link has been severed' (3221225583) with one of ( 'Link has been severed', 'reason(1073741935)' )
 2147483648 : 1073741862
 ../../util/wrap.pl ../../apps/openssl errstr c0000026 => 0
 ok 46 - match 'Socket operation on non-socket' (3221225510) with one of ( 'Socket operation on non-socket', 'reason(1073741862)' )
 2147483648 : 1073741939
 ../../util/wrap.pl ../../apps/openssl errstr c0000073 => 0
 ok 47 - match 'Value too large for defined data type' (3221225587) with one of ( 'Value too large for defined data type', 'reason(1073741939)' )
 2147483648 : 1073741826
 ../../util/wrap.pl ../../apps/openssl errstr c0000002 => 0
 ok 48 - match 'No such file or directory' (3221225474) with one of ( 'No such file or directory', 'reason(1073741826)' )
 2147483648 : 1073741941
 ../../util/wrap.pl ../../apps/openssl errstr c0000075 => 0
 ok 49 - match 'Timer expired' (3221225589) with one of ( 'Timer expired', 'reason(1073741941)' )
 2147483648 : 1073741842
 ../../util/wrap.pl ../../apps/openssl errstr c0000012 => 0
 ok 50 - match 'Invalid cross-device link' (3221225490) with one of ( 'Invalid cross-device link', 'reason(1073741842)' )
 2147483648 : 20
 ../../util/wrap.pl ../../apps/openssl errstr 80000014 => 0
 ok 51 - match '(os/kern) invalid capability' (2147483668) with one of ( '(os/kern) invalid capability', 'reason(20)' )
 2147483648 : 268435468
 ../../util/wrap.pl ../../apps/openssl errstr 9000000c => 0
 ok 52 - match '(ipc/send) invalid memory' (2415919116) with one of ( '(ipc/send) invalid memory', 'reason(268435468)' )
 2147483648 : 1073741855
 ../../util/wrap.pl ../../apps/openssl errstr c000001f => 0
 ok 53 - match 'Too many links' (3221225503) with one of ( 'Too many links', 'reason(1073741855)' )
 2147483648 : 1073741929
 ../../util/wrap.pl ../../apps/openssl errstr c0000069 => 0
 ok 54 - match 'Gratuitous error' (3221225577) with one of ( 'Gratuitous error', 'reason(1073741929)' )
 2147483648 : 268451841
 ../../util/wrap.pl ../../apps/openssl errstr 90004001 => 0
 ok 55 - match '(ipc/rcv) receive in progress' (2415935489) with one of ( '(ipc/rcv) receive in progress', 'reason(268451841)' )
 2147483648 : 1073741930
 ../../util/wrap.pl ../../apps/openssl errstr c000006a => 0
 ok 56 - match 'Invalid or incomplete multibyte or wide character' (3221225578) with one of ( 'Invalid or incomplete multibyte or wide character', 'reason(1073741930)' )
 2147483648 : 21
 ../../util/wrap.pl ../../apps/openssl errstr 80000015 => 0
 ok 57 - match '(os/kern) right exists' (2147483669) with one of ( '(os/kern) right exists', 'reason(21)' )
 2147483648 : 1073741884
 ../../util/wrap.pl ../../apps/openssl errstr c000003c => 0
 ok 58 - match 'Connection timed out' (3221225532) with one of ( 'Connection timed out', 'reason(1073741884)' )
 2147483648 : 1073741943
 ../../util/wrap.pl ../../apps/openssl errstr c0000077 => 0
 ok 59 - match 'Operation canceled' (3221225591) with one of ( 'Operation canceled', 'reason(1073741943)' )
 2147483648 : 1073741858
 ../../util/wrap.pl ../../apps/openssl errstr c0000022 => 0
 ok 60 - match 'Numerical result out of range' (3221225506) with one of ( 'Numerical result out of range', 'reason(1073741858)' )
 2147483648 : 1073741827
 ../../util/wrap.pl ../../apps/openssl errstr c0000003 => 0
 ok 61 - match 'No such process' (3221225475) with one of ( 'No such process', 'reason(1073741827)' )
 2147483648 : 1073741832
 ../../util/wrap.pl ../../apps/openssl errstr c0000008 => 0
 ok 62 - match 'Exec format error' (3221225480) with one of ( 'Exec format error', 'reason(1073741832)' )
 2147483648 : -302
 ../../util/wrap.pl ../../apps/openssl errstr fffffffffffffed2 => 0
 not ok 63 - match 'reason(2147483647)' (18446744073709551314) with one of ( '(ipc/mig) server detected error', 'reason(-302)' )
 
 #   Failed test 'match 'reason(2147483647)' (18446744073709551314) with one of ( '(ipc/mig) server detected error', 'reason(-302)' )'
 #   at ../test/recipes/02-test_errstr.t line 73.
 2147483648 : 1073741825
 ../../util/wrap.pl ../../apps/openssl errstr c0000001 => 0
 ok 64 - match 'Operation not permitted' (3221225473) with one of ( 'Operation not permitted', 'reason(1073741825)' )
 2147483648 : 1073741838
 ../../util/wrap.pl ../../apps/openssl errstr c000000e => 0
 ok 65 - match 'Bad address' (3221225486) with one of ( 'Bad address', 'reason(1073741838)' )
 2147483648 : 1073741904
 ../../util/wrap.pl ../../apps/openssl errstr c0000050 => 0
 ok 66 - match 'Authentication error' (3221225552) with one of ( 'Authentication error', 'reason(1073741904)' )
 2147483648 : 1073741845
 ../../util/wrap.pl ../../apps/openssl errstr c0000015 => 0
 ok 67 - match 'Is a directory' (3221225493) with one of ( 'Is a directory', 'reason(1073741845)' )
 2147483648 : 268435460
 ../../util/wrap.pl ../../apps/openssl errstr 90000004 => 0
 ok 68 - match '(ipc/send) timed out' (2415919108) with one of ( '(ipc/send) timed out', 'reason(268435460)' )
 2147483648 : 268451852
 ../../util/wrap.pl ../../apps/openssl errstr 9000400c => 0
 ok 69 - match '(ipc/rcv) body error' (2415935500) with one of ( '(ipc/rcv) body error', 'reason(268451852)' )
 2147483648 : 268451844
 ../../util/wrap.pl ../../apps/openssl errstr 90004004 => 0
 ok 70 - match '(ipc/rcv) msg too large' (2415935492) with one of ( '(ipc/rcv) msg too large', 'reason(268451844)' )
 2147483648 : 1073741834
 ../../util/wrap.pl ../../apps/openssl errstr c000000a => 0
 ok 71 - match 'No child processes' (3221225482) with one of ( 'No child processes', 'reason(1073741834)' )
 2147483648 : 13
 ../../util/wrap.pl ../../apps/openssl errstr 8000000d => 0
 ok 72 - match '(os/kern) name exists' (2147483661) with one of ( '(os/kern) name exists', 'reason(13)' )
 2147483648 : 1073741828
 ../../util/wrap.pl ../../apps/openssl errstr c0000004 => 0
 ok 73 - match 'Interrupted system call' (3221225476) with one of ( 'Interrupted system call', 'reason(1073741828)' )
 2147483648 : 268451851
 ../../util/wrap.pl ../../apps/openssl errstr 9000400b => 0
 ok 74 - match '(ipc/rcv) header error' (2415935499) with one of ( '(ipc/rcv) header error', 'reason(268451851)' )
 2147483648 : 1073741875
 ../../util/wrap.pl ../../apps/openssl errstr c0000033 => 0
 ok 75 - match 'Network is unreachable' (3221225523) with one of ( 'Network is unreachable', 'reason(1073741875)' )
 2147483648 : 1073741868
 ../../util/wrap.pl ../../apps/openssl errstr c000002c => 0
 ok 76 - match 'Socket type not supported' (3221225516) with one of ( 'Socket type not supported', 'reason(1073741868)' )
 2147483648 : 1073741942
 ../../util/wrap.pl ../../apps/openssl errstr c0000076 => 0
 ok 77 - match 'Not supported' (3221225590) with one of ( 'Not supported', 'reason(1073741942)' )
 2147483648 : 1073741865
 ../../util/wrap.pl ../../apps/openssl errstr c0000029 => 0
 ok 78 - match 'Protocol wrong type for socket' (3221225513) with one of ( 'Protocol wrong type for socket', 'reason(1073741865)' )
 2147483648 : 2505
 ../../util/wrap.pl ../../apps/openssl errstr 800009c9 => 0
 ok 79 - match '(os/device) invalid operation' (2147486153) with one of ( '(os/device) invalid operation', 'reason(2505)' )
 2147483648 : 1073741843
 ../../util/wrap.pl ../../apps/openssl errstr c0000013 => 0
 ok 80 - match 'No such device' (3221225491) with one of ( 'No such device', 'reason(1073741843)' )
 2147483648 : 1073741892
 ../../util/wrap.pl ../../apps/openssl errstr c0000044 => 0
 ok 81 - match 'Too many users' (3221225540) with one of ( 'Too many users', 'reason(1073741892)' )
 2147483648 : 1073741905
 ../../util/wrap.pl ../../apps/openssl errstr c0000051 => 0
 ok 82 - match 'Need authenticator' (3221225553) with one of ( 'Need authenticator', 'reason(1073741905)' )
 2147483648 : 268435469
 ../../util/wrap.pl ../../apps/openssl errstr 9000000d => 0
 ok 83 - match '(ipc/send) no msg buffer' (2415919117) with one of ( '(ipc/send) no msg buffer', 'reason(268435469)' )
 2147483648 : 2
 ../../util/wrap.pl ../../apps/openssl errstr 80000002 => 0
 ok 84 - match '(os/kern) protection failure' (2147483650) with one of ( '(os/kern) protection failure', 'reason(2)' )
 2147483648 : 1073741933
 ../../util/wrap.pl ../../apps/openssl errstr c000006d => 0
 ok 85 - match 'Multihop attempted' (3221225581) with one of ( 'Multihop attempted', 'reason(1073741933)' )
 2147483648 : -300
 ../../util/wrap.pl ../../apps/openssl errstr fffffffffffffed4 => 0
 not ok 86 - match 'reason(2147483647)' (18446744073709551316) with one of ( '(ipc/mig) client type check failure', 'reason(-300)' )
 
 #   Failed test 'match 'reason(2147483647)' (18446744073709551316) with one of ( '(ipc/mig) client type check failure', 'reason(-300)' )'
 #   at ../test/recipes/02-test_errstr.t line 73.
 2147483648 : 1073741886
…
```
Looking at it with more detail, it appears that the return from @Errno::EXPORT_OK is wrong.
My understanding is that you attempt to iterate over errno codes from 1 to X (EPERM, ENOENT, ...)
Hurd wise it would be 
```

EPERM  -> 0x40000001
ENOENT -> 0x40000002

```
With a 0x40000000 prefix (based on my observation).  The perl code
```
 foreach my $errname (@Errno::EXPORT_OK) {
    my $errcode = "Errno::$errname"->();
    printf "Code: %x %s\n", $errcode, $errname;
 }

```
Returns:
```
Code: 9c7 ED_ALREADY_OPEN
Code: 40000079 ENOTRECOVERABLE
Code: 4000003b ETOOMANYREFS
Code: 4000004c EPROCUNAVAIL
Code: 9c9 ED_INVALID_OPERATION
Code: 4000000a ECHILD
Code: 4000006d EMULTIHOP
Code: 40000078 EOWNERDEAD
Code: 40000064 EBACKGROUND
Code: 40000040 EHOSTDOWN
Code: 4000003f ENAMETOOLONG
Code: 40000033 ENETUNREACH
Code: 4000004e ENOSYS
Code: 40000020 EPIPE
Code: 40000070 ENOMSG
Code: 40000034 ENETRESET
Code: 40000067 EGREGIOUS
Code: fffffffffffffecb EMIG_DESTROY_REQUEST
Code: 10000007 EMACH_SEND_INTERRUPTED
Code: 40000042 ENOTEMPTY
Code: 40000023 EAGAIN
Code: 10000003 EMACH_SEND_INVALID_DEST
Code: 40000050 EAUTH
Code: 14 EKERN_INVALID_CAPABILITY
Code: 1b EKERN_TIMEDOUT
Code: 4000004b EPROGMISMATCH
Code: 40000015 EISDIR
Code: 40000014 ENOTDIR
Code: 10000008 EMACH_SEND_MSG_TOO_SMALL
Code: 10000004 EMACH_SEND_TIMED_OUT
Code: 13 EKERN_UREFS_OVERFLOW
Code: 40000010 EBUSY
Code: c EKERN_NOT_IN_SET
Code: 40000051 ENEEDAUTH
Code: 12 EKERN_INVALID_VALUE
Code: 40000068 EIEIO
Code: 4000002d EOPNOTSUPP
Code: 40000011 EEXIST
Code: 4000000f ENOTBLK
Code: 4000002b EPROTONOSUPPORT
Code: 10000006 EMACH_SEND_NOTIFY_IN_PROGRESS
…
```
The codes with wth 4 prefix appears correct. The one with the 1 prefix are EMACH_ something, different namespace. The 0 zero prefix looks like kernel.

**Comment by levitte** [comment]
@errno::EXPORT_OK is the list of all the error codes that the perl installation has been able to find in the system.  It seems, though, that some of those error codes are interpreted as negative numbers by perl, and that throws off the test.  In your last comment, this is one of them:

```
Code: fffffffffffffecb EMIG_DESTROY_REQUEST
```

and indeed, there seems to be a set of negative error codes, at least coming with GNU libc:

https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/mach/hurd/bits/errno.h;h=8f2fbfd80ffd3dc3c765a7635203d69b1329ead4;hb=HEAD#l199

I'm frankly not sure if it's worth the while trying to accomodate them, or if the 02-test_errstr.t should simply ignore negative error codes.  From looking at them, it seems very improbable that they will end up in an OpenSSL error report...

**Comment by t8m** [comment]
+1 to just ignore the negative error codes.

**Comment by levitte** [comment]
Fix in #14779.  @sebastianas, would you mind trying it out?

**Comment by sebastianas** [comment]
It passes the testsuite. Thank you.
